### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by marijnvanderhorst**
## Proposed changes
Implements bare minimum implementation to add login and logout routes to any Laravel application.

## Types of changes

What types of changes does your code introduce to this repository?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Testing
As always

### Unit testing
Phpunit

### Manual testing
* Add this as dependency to a Laravel installation.
* Use Auth0 login provider
* Verify login and logout works.

## Further comments
n/a

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+os: linux
+language: php
+dist: bionic
+php:
+  - 7.3
+  - 7.4
+install:
+  - composer install
+script:
```
Comment (EmielSchmeink) [inline .travis.yml:9]: I think we should add the linter stuff as well, since I would like to not have a PSM deja vu.

Comment (marijnvanderhorst) [reply .travis.yml:9]: ok ok will add a linter

Comment (EmielSchmeink) [reply .travis.yml:9]: Nvm, this is the auto0 package, I care less now.

Comment (marijnvanderhorst) [reply .travis.yml:9]: Added it anyway

File: phpmd.xml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        using http://phpmd.org/documentation/creating-a-ruleset.html
+    </description>
+    <rule ref="rulesets/cleancode.xml">
+        <exclude name="StaticAccess"/>
+        <exclude name="ElseExpression"/>
+        <exclude name="BooleanArgumentFlag"/>
+    </rule>
+    <rule ref="rulesets/codesize.xml/CyclomaticComplexity">
+        <properties>
+            <property name="reportLevel" value="20" />
```
Comment (EmielSchmeink) [inline phpmd.xml:19]:
```suggestion
            <property name="reportLevel" value="10" />
```

File: phpmd.xml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        </properties>
+    </rule>
+    <rule ref="rulesets/codesize.xml/NPathComplexity">
+        <properties>
+            <property name="minimum" value="1100" />
```
Comment (EmielSchmeink) [inline phpmd.xml:24]:
```suggestion
            <property name="minimum" value="200" />
```

File: phpmd.xml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        </properties>
+    </rule>
+    <rule ref="rulesets/codesize.xml/ExcessiveMethodLength">
+        <properties>
+            <property name="minimum" value="150" />
```
Comment (EmielSchmeink) [inline phpmd.xml:29]:
```suggestion
            <property name="minimum" value="100" />
```

File: phpmd.xml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        </properties>
+    </rule>
+    <rule ref="rulesets/codesize.xml/ExcessiveClassLength"/>
+    <rule ref="rulesets/codesize.xml/ExcessiveParameterList"/>
+    <rule ref="rulesets/codesize.xml/ExcessivePublicCount"/>
+    <rule ref="rulesets/codesize.xml/TooManyFields"/>
+    <rule ref="rulesets/codesize.xml/TooManyMethods">
+        <properties>
+            <property name="maxmethods" value="30"/>
```
Comment (EmielSchmeink) [inline phpmd.xml:38]:
```suggestion
            <property name="maxmethods" value="25"/>
```

File: phpmd.xml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        </properties>
+    </rule>
+    <rule ref="rulesets/codesize.xml/ExcessiveClassComplexity">
+        <properties>
+            <property name="maximum" value="75"/>
```
Comment (EmielSchmeink) [inline phpmd.xml:43]:
```suggestion
            <property name="maximum" value="50"/>
```

File: phpmd.xml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        </properties>
+    </rule>
+
+    <rule ref="rulesets/controversial.xml"/>
+    <rule ref="rulesets/design.xml/CouplingBetweenObjects">
+        <properties>
+            <property name="maximum" value="25"/>
```
Comment (EmielSchmeink) [inline phpmd.xml:50]:
```suggestion
            <property name="maximum" value="13"/>
```

**Review by EmielSchmeink — APPROVED** [review]

**Pull request merged**